webapp/src/ts/modules/contacts/contacts-content.component.ts

```diff
@@ -238,14 +239,17 @@ export class ContactsContentComponent implements OnInit, OnDestroy {
 
     const allReports = reports || this.selectedContact?.reports || [];
     this.filteredReports = allReports
-      .filter((report) => !reportStartDate || reportStartDate.isBefore(report.reported_date));
+      .filter((report) => !reportStartDate || reportStartDate.isBefore(report.reported_date))
+      .slice(0, this.DISPLAY_LIMIT);
```

Comment:
This is just a note that since before [we were displaying only 50 records](https://github.com/medic/cht-core/blob/c4c3114ed8e17f49a78e1f803a643792b71f2c8e/webapp/src/ts/services/search.service.ts#L133) in all the tabs (3m, 6m, view all)

Reply:
@Benmuiruri, thinking about this more, it would help in performance if it stops the loop when having 50 reports in the date range, instead of processing 500 reports. 

```
newFilteredReports = [];
allReports.forEach(report => {
  if (!reportStartDate || reportStartDate.isBefore(report.reported_date)) {
     newFilteredReports.push(report)
  }
  if (filteredReports > this.DISPLAY_LIMIT) {
     return;
  }
);

this.filteredReports = newFilteredReports

```